File: Aula4/Exercicios-aula4/projeto_circ_cil/src/main/java/aula4/projeto_circ_cil/Cilindro.java
package aula4.projeto_circ_cil;
import javax.swing.JOptionPane;


public class Cilindro {
    private float altura;
    private Circulo base;
    
    public Cilindro(){}
    
    public Cilindro(float altura, Circulo base){
        this.altura = altura;
        this.base = new Circulo(base.raio);
    }
    
    public float calculaArea(){
        float lateral = (this.base.pi * 2) * (this.base.raio * altura);
        float areaTotal = (2 * this.base.calculaArea()) + (2 * lateral);
        return areaTotal;
    }
    
    public float calculaVolume(){
        return this.altura * this.base.calculaArea();
    }
    
    public void imprimeDados(){
    if(altura > 0.0f){
        JOptionPane.showMessageDialog(null, "A área do cilindro com altura " +
                this.altura + 
                " é: " + calculaArea() +
                "\n O volume deste cilíndro é: " + 
                calculaVolume());
    }
    }
}

File: CsharpStyle/Aula4/Ex1/classes/Cilindro.cs
namespace Ex1;

public class Cilindro
{
    private float _altura;
    private Circulo _base;

    public Cilindro() { }
    public Cilindro(float altura, Circulo base1)
    {
        _altura = altura;
        _base = new Circulo(base1.Raio);
    }

    public float calculaArea()
    {
        float lateral = (_base.Pi * 2) * (_base.Raio * _altura);
        float areaTotal = (2 * _base.calculaArea()) + (2 * lateral);
        return areaTotal;
    }

    public float calculaVolume()
    {
        return _altura * _base.calculaArea();
    }

    public override string ToString()
    {
        if (_altura > 0.0f)
        {

            return $" A área do cilindro com altura {_altura.ToString()} é: {calculaArea().ToString()}\nO volume deste cilindro é: {calculaVolume()}";
        }
        return "Cilindro inválido";
    }
}
